app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -0,0 +1,81 @@
+module Katello
+  class Api::V2::HostSubscriptionsController < Katello::Api::V2::ApiController
+    respond_to :json
+
+    before_filter :find_host_with_subscriptions
+
+    api :GET, "/hosts/:id/subscriptions", N_("List a host's subscriptions"), :deprecated => true
+    param :id, String, :desc => N_("id of the host"), :required => true
+    def index
+      @collection = index_response
+      respond_for_index :collection => @collection
+    end
+
+    def index_response
+      entitlements = @candlepin_consumer.entitlements
+      subscriptions =  entitlements.map { |entitlement| ::Katello::SystemSubscriptionPresenter.new(entitlement) }
+      full_result_response(subscriptions)
+    end
+
+    api :PUT, "/hosts/:id/subscriptions/auto_attach", N_("Trigger an auto-attach of subscriptions")
+    param :id, String, :desc => N_("id of the host"), :required => true
+    def auto_attach
+      sync_task(::Actions::Katello::Host::AutoAttachSubscriptions, @host)
+      respond_for_index(:collection => index_response, :template => "index")
```

Comment:
It does not actually, a better name for it would probably be respond_for_collection() but those are based on the action name and current depend on that.  Its not ideal